src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -190,11 +191,15 @@ Pipe ReadFromMergeTree::readInOrder(
     bool use_uncompressed_cache)
 {
     Pipes pipes;
+    /// For reading in order it makes sense to read only
+    /// one range per task to reduce number of read rows.
+    bool one_range_per_task = read_type != ReadType::Default;
+
```

Comment:
Is this relevant only for a case when we have a LIMIT and won't read all the data? Looks like it should make no difference for a full scan...

Reply:
Yes. I've added a pushdown of limit.